frontend/app/controllers/restaurant-list.js

```diff
@@ -0,0 +1,124 @@
+import Controller from "@ember/controller";
+import { inject as service } from "@ember/service";
+import { isArray } from "@ember/array";
+import CONSTANTS from "../constants";
+import { set } from "@ember/object";
+import EmberObject from "@ember/object";
+
+export default Controller.extend({
+  _reviewService: service("review-service"),
+  _swalService: service("swal-service"),
+  _restaurantService: service("restaurant-service"),
+  router: service("-routing"),
+  name: "",
+  cousineFilters: [],
+  range: "",
+  rating: "",
+  searchParameters: "",
+  sortAsc: false,
+  checkedKey: true,
+  sortKey: "",
+  pageSize: CONSTANTS.POPULAR_RESTAURANTS_SIZE,
+  pageNumber: 1,
+  numOfPages: [],
+  pageMargin: "",
+  activePage: EmberObject.create({
+    number: 1,
+    activeStatus: true
+  }),
+
+  addParameter(paramName, paramValue) {
+    let searchParam = this.get("searchParameters");
+    searchParam += paramName;
+    searchParam += "=";
+    if (isArray(paramValue)) {
+      for (let i = 0; i < paramValue.length; i++) {
+        searchParam += paramValue[i] + ","; // comma on last value won't make a difference
+      }
+    } else {
+      searchParam += paramValue;
+    }
+    searchParam += "&";
+    this.set("searchParameters", searchParam);
+  },
+
+  actions: {
+    hideDropDown(element) {
+      document.getElementById(element).classList.remove("show-drop");
+    },
+    showDropDown(element) {
+      document.getElementById(element).classList.add("show-drop");
+    },
+    changeCousineStatus(cousine) {
+      let cousineFilters = this.get("cousineFilters");
+      if (cousine.activeStatus) {
+        set(cousine, "activeStatus", false);
+        // removing cousine name from filter
+        let indexOfElement = cousineFilters.indexOf(cousine.name);
```

Comment:
https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore read that article, for js functions.. here you can use cousineFilters.find(filter => filter === cousine.name);